x-pack/plugin/ent-search/src/test/java/org/elasticsearch/xpack/application/connector/ConnectorFilteringTests.java

```diff
@@ -110,6 +110,73 @@ public void testToXContent() throws IOException {
 
     }
 
+    public void testToXContent_WithAdvancedSnippetPopulated() throws IOException {
```

Comment:
Similar suggestion to the one in the integration tests. Should we've explicit test for the different cases: literal, object and array?